lib/hammer_cli_foreman/commands.rb

```diff
@@ -328,7 +328,9 @@ def pagination_supported?
     end
 
     def should_retrieve_all?
-      pagination_supported? && option_per_page.nil? && option_page.nil? && HammerCLI::Settings.get(:ui, :per_page).nil?
+      retrieve_all = (pagination_supported? && option_per_page.nil? && option_page.nil?)
+      retrieve_all = (retrieve_all && HammerCLI::Settings.get(:ui, :per_page).nil?) if output.adapter.paginate_by_default?
```

Comment:
Just a style thing - it could be shortened to:

```ruby
retrieve_all &&= HammerCLI::Settings.get(:ui, :per_page).nil? if output.adapter.paginate_by_default?
```